src/utilities/geometry/Intersection.cpp

```diff
@@ -64,10 +65,56 @@ namespace openstudio {
 
 // Private implementation functions
 
+// Cleans a polygon by shrinking and expanding
+BoostPolygon removeSpikesEx(const BoostPolygon& polygon) {
+
+  //const double buffer_distance = 1.0;
+  //const int points_per_circle = 36;
+  const double amount = 0.005;
+
+  boost::geometry::strategy::buffer::distance_symmetric<coordinate_type> expand(amount);
+  boost::geometry::strategy::buffer::distance_symmetric<coordinate_type> shrink(-amount);
+  boost::geometry::strategy::buffer::join_miter join_strategy;
+  boost::geometry::strategy::buffer::end_flat end_strategy;
+  boost::geometry::strategy::buffer::side_straight side_strategy;
+  boost::geometry::strategy::buffer::point_circle point_strategy;
+
+  // cppcheck-suppress constStatement
+  BoostMultiPolygon resultExpand;
+  // cppcheck-suppress constStatement
+  BoostMultiPolygon resultShrink;
+  // cppcheck-suppress constStatement
+  BoostMultiPolygon result;
+  boost::geometry::buffer(polygon, resultShrink, shrink, side_strategy, join_strategy, end_strategy, point_strategy);
+  boost::geometry::buffer(resultShrink, resultExpand, expand, side_strategy, join_strategy, end_strategy, point_strategy);
+  boost::geometry::simplify(resultExpand, result, amount);
+
+  if (result.size() == 0)
+    return polygon;
+  else {
+    // The returned points are adjusted to the input polygon (which defines the canonical set)
+    for (unsigned i = 0; i < result[0].outer().size(); ++i) {
+      Point3d point3d(result[0].outer()[i].x(), result[0].outer()[i].y(), 0.0);
+
+      // Outer ring of the original polygon
+      for (unsigned j = 0; j < polygon.outer().size(); j++) {
+        Point3d p1(polygon.outer()[j].x(), polygon.outer()[j].y(), 0);
+        // Two points are within tolerance set the result to the original input point
+        if (getDistance(point3d, p1) <= 0.05) {
+          result[0].outer()[i].x(polygon.outer()[j].x());
+          result[0].outer()[i].y(polygon.outer()[j].y());
+          break;
+        }
+      }
+    }
+    return result[0];
+  }
+}
+
 BoostPolygon removeSpikes(const BoostPolygon& polygon) {
   BoostPolygon temp(polygon);
   boost::geometry::remove_spikes(temp);
-  return temp;
+  return removeSpikesEx(temp);
```

Comment:
Do you need both remove spikes calls?

Reply:
I didn't see any harm in leaving the original call in so I left it. That way there's two attempts to remove spikes.

Maybe in the future the boost call can be removed